### PR TITLE
Fix Overview page to convey general idea of project

### PR DESCRIPTION
**Pull request by bchen28**
The main purpose of the Overview page should be giving new users a general idea of what the Tremor project is and what it can be used for. According the Tech Assessments Doc written by Celeste Horgon, I hope to coalesce general information onto this page from the original Architecture Overview page and the About us page to give users a more well-rounded idea of what Tremor is. Moving forward, I want to suggest a cross-section of documentation hierarchy that will allow for easier cross-navigation between pages and understanding of essential terms. 
- Docs 
    - Overview 
    - Model 
        - Onramps
        - Offramps
        - Pipeline
        - Codecs
    - etc. 

I've left most of the more esoteric model information in this page for now, but my next step will be removing that material into separate pages and subsections.

### CODE REVIEW

**Comment by darach** [comment]
The terms `onramp` and `offramp` are legacy terms. We are transitioning to the terms `source`, `sink` and `connector` here as they are more general terms. The `connector` branch ( and related RFC ) is a work in progress but the legacy terms will be written out of documentation in the next ( v0.12 ) release of the project.

**Review by darach — APPROVED** [review]
LGTM 👍

File: docs/overview.md [hunk 1]
```diff
@@ -4,28 +4,30 @@ title: Architecture Overview
...
+- Data Bridging - Tremor bridges asynchronous sources with synchronous sinks. Its ability to bridge sync/async and vice-versa enables message flows to be classified, dimensioned, segmented, and routed using user defined logic while handling back-pressure and other distributed systems problems on behalf of its operators.
+- Distribution at Scale - Tremor is designed for high volume messaging environments. Using proactive rate limiting and a novel algorithm to handle back-propagation, Tremor optimizes back-pressure handling at capacity up to 10x more efficiently than alternatives such as Logstash and Telegraf.
+- In-flight Redeployment - Tremor can be reconfigured via its API to allow workload migration and reconfiguration without redeployment.
+- Event processing - Tremor adopts principles from the Distributed Event Based Systems, Event Stream Processor (ESP) and Complex Event Processing (CEP) communities; however, it currently encompasses an incomplete feature set. Over time, Tremor may evolve as an ESP or CEP solution, but this is not an explicit goal for the project.
```
Comment (skoech) [inline docs/overview.md:22]: You might want to add the 'DEBS' abbreviation like for the others here.

File: docs/overview.md [hunk 2]
```diff
@@ -161,7 +156,7 @@ These terms can be used interchangeably.
 
 Operators CAN conduct or orchestrate one or many Tremor servers through its REST based API.
 
-The API in turn interfaces with registry and repository facilities. Tremor distinguishes between artefacts and instances. Artefacts in Tremor have no runtime overhead.
+The API in turn interfaces with registry and repository facilities. Tremor distinguishes between artefacts and instances. artefacts in Tremor have no runtime overhead.
```
Comment (skoech) [inline docs/overview.md:159]: Not sure why you changed the case for 'Artefacts' here.

File: docs/overview.md [hunk 3]
```diff
@@ -170,7 +165,7 @@ Artefacts in Tremor are declarative specifications of:
 - Pipelines - A pipeline specification is a specific configuration of a pipeline graph
 - Bindings - A binding specification describes how onramps, pipelines and offramps should be interconnected
 
-Artefacts can be thought of analagously to code. They are a set of instructions, rules or configurations. As such they are registered with Tremor via its API and stored in Tremor's artefact repository.
+Artefacts can be thought of analagous to code. They are a set of instructions, rules or configurations. As such they are registered with Tremor via its API and stored in Tremor's artefact repository.
```
Comment (skoech) [inline docs/overview.md:168]: How about "Artefacts can be thought of as analagous to code."?

File: docs/overview.md [hunk 4]
```diff
@@ -170,7 +165,7 @@ Artefacts in Tremor are declarative specifications of:
...
 
 Deployment in Tremor, is achieved through a mapping artefact. The mapping artefact specifies how artefacts should be deployed into one or many runtime instances, activated, and connected to live instances of onramps or offramps.
```
Comment (skoech) [inline docs/overview.md:170]: "Deployment in Tremor is achieved through a mapping artefact." No need for the comma, right? :D

**Review by skoech — CHANGES_REQUESTED** [review]
Again, absolutely brilliant. Brief and direct in a refreshing way. Maybe just a few changes I thought you could go over.

**Comment by bchen28** [comment]
I turned this into a rather exhaustive content overhaul of what preexisted in the Overview page. No content was moved elsewhere, existing information was simply consolidated, reorganized, and rewritten to provide a more concise and user-friendly understanding of the project. I'm too tired and it's a bit late in the night for me to comment on specific questions I had regarding wording and technical accuracy, but I can link those tomorrow. Would appreciate any thoughts/comments @skoech @darach @Licenser

File: docs/overview.md [hunk 5]
```diff
@@ -19,34 +19,31 @@ Tremor is designed to handle the following:
...
 - Distribution at Scale - Tremor is designed for high volume messaging environments. Using proactive rate limiting and a novel algorithm to handle back-propagation, Tremor optimizes back-pressure handling at capacity up to 10x more efficiently than alternatives such as Logstash and Telegraf.
 - In-flight Redeployment - Tremor can be reconfigured via its API to allow workload migration and reconfiguration without redeployment.
-- Event processing - Tremor adopts principles from the Distributed Event Based Systems (DEBS), Event Stream Processor (ESP) and Complex Event Processing (CEP) communities; however, it currently encompasses an incomplete feature set. Over time, Tremor may evolve as an ESP or CEP solution, but this is not an explicit goal for the project.
+- Event processing - Tremor's primary goal is to operate as an Extract, Transform, and Loading (ETL) tool. While Tremor adopts principles from Distributed Event Based Systems (DEBS), Event Stream Processor (ESP) and Complex Event Processing (CEP) infrastructures, it is by no means an exhaustive system that encompasses all the aforementioned communities' features and requirements.
```
Comment (skoech) [inline docs/overview.md:22]: This is a great synopsis of how @darach explained the Tremor architecture and functionality yesterday. But, I was thinking maybe we should maintain the element of possibility here. Like, "It's currently not exhaustive..." or, "It is by no means an exhaustive system... yet...". Something along those lines. What do you think?

File: docs/overview.md [hunk 6]
```diff
@@ -19,34 +19,31 @@ Tremor is designed to handle the following:
...
+| `tremor:///binding/bn_01/tl_01` | An active instance of data exchange between two artefacts                                         |
 |                              |                                                                                                            |
 
-The Tremor REST API and configuration file formats also follow the same URL format.
-
-In the case of configuration, a shorthand URL form is often used. In the configuration model, we discriminate artefacts by type, so it is often sufficient to infer the `tremor:///{artefact-kind}` component when specifying ( configuring ) artefacts.
+The Tremor REST API and configuration file formats follow the same URL format.
 
-In bindings, however, we need to use the full URL path component ( for example:`/pipeline/01` ).
+The current configuration model uses a shorthand URL form. It discriminates artefacts by type, so it is often sufficient to infer the `tremor:///{artefact-kind}` component when specifying configuring artefacts.
```
Comment (skoech) [inline docs/overview.md:44]: ... specifying or configuring artefacts.

File: docs/overview.md [hunk 7]
```diff
@@ -19,34 +19,31 @@ Tremor is designed to handle the following:
...
 
-At this time, the full URL form is not used in the configuration model.
+For bindings, full URL path components are needed (ie.:`/pipeline/01`).
```
Comment (skoech) [inline docs/overview.md:46]: I think e.g. is more appropriate here.

File: docs/overview.md [hunk 8]
```diff
@@ -61,156 +58,141 @@ Note: The Processing model will likely evolve over time alongside concurrency, t
...
+Tremor imposes causal event ordering over ingested events and processes events deterministically, although, given its nature as a distributed system, it cannot impose a total ordering over all ingested events.
 
-Requests from multiple independent sources over the same pipeline may arbitrarily interleave, but should not re-order.
+Events flowing into Tremor from multiple onramps and multiple clients are considered independent. However, events sent by a _specific_ client through a _specific_ onramp into a _passthrough_ pipeline will flow through Tremor in their _origin_ order and be passed to offramps in the same _origin_ order. Requests from multiple independent sources over the same pipeline may arbitrarily interleave, but will not re-order.
 
-In pipelines, events are processed in depth first order. Where Tremor operators have no intrinsic ordering ( such as a branch split ), Tremor internally _imposes_ an order.
+In pipelines, events are processed in depth first order. In cases where Tremor operators have no intrinsic ordering (such as a branch split), Tremor internally _imposes_ an order.
```
Comment (skoech) [inline docs/overview.md:68]: depth-first

File: docs/overview.md [hunk 9]
```diff
@@ -61,156 +58,141 @@ Note: The Processing model will likely evolve over time alongside concurrency, t
...
 Artefacts in Tremor are declarative specifications of:
 
-- Onramps - An onramp specification is a specific configuration of a supported onramp kind
-- Offramps - An offramp specification is a specific configuration of a supported offramp kind
-- Pipelines - A pipeline specification is a specific configuration of a pipeline graph
-- Bindings - A binding specification describes how onramps, pipelines and offramps should be interconnected
+- Onramps - a specific configuration of a supported onramp type
+- Offramps - a specific configuration of a supported offramp type
+- Pipelines - a specific configuration of a pipeline graph
+- Bindings - a specefic configuration of how all other artefacts should be connected
```
Comment (skoech) [inline docs/overview.md:155]: specific*

File: docs/overview.md [hunk 10]
```diff
@@ -61,156 +58,141 @@ Note: The Processing model will likely evolve over time alongside concurrency, t
...
 
-In general, operators and developers should _minimize_ the number of encoding and decoding steps required in the transit of data through Tremor or between Tremor instances.
-
-The major overhead in most Tremor systems is encoding and decoding overhead. To compensate that, as JSON is the most dominant format, we [ported](https://github.com/simd-lite/simdjson-rs) [simd-json](https://github.com/lemire/simdjson) this reduces the cost of en- and decoding significantly compared to other JSON implementations in Rust.
+Since the major overhead in most Tremor systems is encoding and decoding and JSON is the dominant data format, we [ported](https://github.com/simd-lite/simdjson-rs) [simd-json](https://github.com/lemire/simdjson) to reduce the cost of encoding and decoding significantly compared to other JSON implementations in Rust.
 
 ### Distribution model
 
-Tremor does not ( yet ) have an out-of-the-box network protocol. A native Tremor protocol is planned in the immediate / medium term.
+Tremor does not yet have an out-of-the-box network protocol. A native Tremor protocol is planned for development in the immediate/mid-term.
```
Comment (skoech) [inline docs/overview.md:189]: I think medium-term is more appropriate as it references a  period of time in the near future.

**Review by skoech — APPROVED** [review]
Looks great to me. 👍🏿

**Review by Licenser — APPROVED** [review]
re apprive since update unapproved

**Pull request merged**